sys/vfs_syscalls.c

```diff
@@ -173,3 +200,21 @@ int sys_fstat(thread_t *td, syscall_args_t *args) {
     return error;
   return 0;
 }
+
+int sys_dup(thread_t *td, syscall_args_t *args) {
+  int old = args->args[0];
+  int *new = (int *)(uintptr_t)args->args[1];
```

Comment:
Do you really trust user-space program to pass correct pointer? ;-) Please use [fuword32 & suword32](http://mimiker.ii.uni.wroc.pl/source/xref/mimiker/include/systm.h#15) calls.